### ISSUE TITLE
API connector: wizard is stuck on Basic auth is used

### ISSUE THREAD

**Issue by dsimansk** [syndesisio/syndesis#944]
Our [todo-example](https://github.com/syndesisio/todo-example/blob/master/swagger.json#L173-L177) defines Basic auth in swagger that causes some trouble to the UI wizard.

Browser console error
```
vendor.7ec7098e6313e0e15893.bundle.js:1 ERROR TypeError: Cannot read property 'defaultValue' of undefined
    at n.ngOnInit (2.627bf80f6122cd2a77ea.chunk.js:1)
    at Cn (vendor.7ec7098e6313e0e15893.bundle.js:1)
    at dr (vendor.7ec7098e6313e0e15893.bundle.js:1)
    at cr (vendor.7ec7098e6313e0e15893.bundle.js:1)
    at jr (vendor.7ec7098e6313e0e15893.bundle.js:1)
    at Object.updateDirectives (2.627bf80f6122cd2a77ea.chunk.js:1)
    at Object.updateDirectives (vendor.7ec7098e6313e0e15893.bundle.js:1)
    at ur (vendor.7ec7098e6313e0e15893.bundle.js:1)
    at Cr (vendor.7ec7098e6313e0e15893.bundle.js:1)
    at wr (vendor.7ec7098e6313e0e15893.bundle.js:1)
    at ur (vendor.7ec7098e6313e0e15893.bundle.js:1)
    at Cr (vendor.7ec7098e6313e0e15893.bundle.js:1)
    at _r (vendor.7ec7098e6313e0e15893.bundle.js:1)
    at ur (vendor.7ec7098e6313e0e15893.bundle.js:1)
    at Cr (vendor.7ec7098e6313e0e15893.bundle.js:1)
```

IMHO the UI code is looking for `authenticationType.defaultValue` && `authorizationEndpoint.defaultValue` but in case of Basic Auth, only the authenticationType is present.

https://github.com/syndesisio/syndesis/blob/537727fa366179fa8e1ddad9c63b6f617cf71ab3/app/ui/src/app/customizations/api-connector/api-connector-create/api-connector-auth/api-connector-auth.component.ts#L21-L27

syndesis-rest response for `/connectors/custom/info` query
```
"authenticationType": {
      "componentProperty": true,
      "defaultValue": "basic",
      "deprecated": false,
      "description": "Type of authentication used to connect to the API",
      "displayName": "Authentication Type",
      "group": "producer",
      "javaType": "java.lang.String",
      "kind": "property",
      "label": "producer",
      "required": false,
      "secret": false,
      "type": "string",
      "tags": [
        "authentication-type"
      ],
      "enum": [
        {
          "label": "HTTP Basic Authentication",
          "value": "basic"
        }
      ]
    },
```

**Comment by deeleman** [comment]
> IMHO the UI code is looking for `authenticationType.defaultValue && authorizationEndpoint.defaultValue`. 

You're right. This is how it was implemented, according to the spes provided. It's obvious that this step will require some rethinking then.

**Comment by dsimansk** [comment]
Fixed via #946, thanks @deeleman 👍
